src/react/components/common/tagInput/tagInput.tsx

```diff
@@ -613,16 +643,57 @@ export class TagInput extends React.Component<ITagInputProps, ITagInputState> {
     private getTypeSubMenuItems = (): IContextualMenuItem[] => {
         const tag = this.state.selectedTag;
         const types = Object.values(FieldType);
+        // If it's an old existing tag (before 'assigned' been introduced) -> update tag to new format
+        if (tag && tag.assigned === undefined) {
+            const newTag = {
+                ...tag,
+                assigned: true,
+            };
+            this.props.onTagChanged(tag, newTag);
+            types.map((type) => {
+                return {
+                    key: type,
+                    text: type,
+                    canCheck: true,
+                    isChecked: type === tag.type,
+                    onClick: this.onTypeSelect,
+                } as IContextualMenuItem;
+            });
+        }
+        // New tag - user can assign any type
+        if (tag.assigned === false) {
```

Comment:
`var === false` or `var === true` is never needed  
just`!var` or `var`
it's better coding practice

Reply:
@stew-ro  For the new tags value is explicitly set as "false", for existing tags value it is 'undefined' which is falsy value as well...

Reply:
yeah, but you already checked for undefined, so you could just do !tag.assigned here